src/librustdoc/clean/mod.rs

```diff
@@ -793,9 +836,171 @@ impl AttributesExt for Attributes {
     }
 }
 
+enum PathKind {
+    /// can be either value or type, not a macro
+    Unknown,
+    /// macro
+    Macro,
+    /// values, functions, consts, statics, everything in the value namespace
+    Value,
+    /// types, traits, everything in the type namespace
+    Type
+}
 impl Clean<Attributes> for [ast::Attribute] {
     fn clean(&self, cx: &DocContext) -> Attributes {
-        Attributes::from_ast(cx.sess().diagnostic(), self)
+        let mut attrs = Attributes::from_ast(cx.sess().diagnostic(), self);
+
+        if UnstableFeatures::from_environment().is_nightly_build() {
```

Comment:
This should probably be checking for a feature flag or command line option.

Reply:
There's actually another way to check if it's a nightly build or not iirc. Maybe better?

Reply:
This line was stolen from the bit where it parses error numbers in doctests. Rustdoc can't take feature flags of its own, and i didn't want to add a rustc feature flag for it since there's nothing for rustc itself to check. A command-line flag could work, but it feels unwieldy for something that will cause someone's docs to work or not work entirely. At least, that was the reasoning that led me to make it "globally for nightly".

Reply:
I did it for some unstable flags. It can found be pretty easily since I removed it recently. I'll give you the line when I have time to look for it!

Reply:
Ok so I found this: https://github.com/rust-lang/rust/blob/master/src/librustdoc/lib.rs#L289

Maybe it's what we were looking for?